src/frontend/components/render-state/render-state.ts

```diff
@@ -40,14 +80,19 @@ export class RenderState {
     return false;
   }
 
-  expandTree(key, $event) {
-    this.expansionState[key] = !this.expansionState[key];
-
-    $event.preventDefault();
-    $event.stopPropagation();
+  private classification(key: string): StateClassification {
+    if (this.level === 0) {
+      if (this.inputs.hasOwnProperty(key)) {
+        return StateClassification.Input;
+      }
+      else if (this.outputs.hasOwnProperty(key)) {
+        return StateClassification.Output;
```

Comment:
Maybe these could be queried separately. There is no case where a property name at level 0 can be both `@Input` and `@Output` so you might as well test them separately in the template.